common/temppath/src/lib.rs

```diff
@@ -28,11 +28,16 @@ impl Drop for TempPath {
 }
 
 impl TempPath {
-    /// Create new uninitialized temporary path, i.e. a file or directory isn't created
-    /// automatically
+    /// Create new uninitialized temporary path in the system temp directory, note: a file /
+    /// directory isn't created automatically
     pub fn new() -> Self {
-        // Create a random path in the system temp directory
-        let mut temppath = std::env::temp_dir();
+        Self::new_with_temp_dir(std::env::temp_dir())
+    }
+
+    /// Create new uninitialized temporary path in the specified directory, note: a file /
+    /// directory isn't created automatically
```

Comment:
Perhaps, this statement will be less ambiguous.

/// Create new uninitialized temporary path in the specified directory, note that 
/// the temp_dir directory isn't created automatically